### PR TITLE
#52: Added CLI options for specifying the timeout for the ExtractValidator

### LINKED ISSUE #52: SLC Deployer: Add timeout option to CLI
In the scope of ticket #50 a new `ExtractValidator` has been implemented.

The current ticket requests adding options to the CLI so that an interactive user can control the timeout duration.

Proposal: 
```python
@click.option('--extract-timeout-minutes', type=int, default=10)
```

Open question:
* Should the CLI specify a default-callback?
* Should the CLI enable to select a callback?

### PR DESCRIPTION
**Pull request by ckunki**
Closes #52

### CODE REVIEW

File: exasol/python_extension_common/deployment/language_container_deployer.py [hunk 1]
```diff
@@ -306,7 +310,10 @@ def create(cls,
                path_in_bucket: str = '',
                use_ssl_cert_validation: bool = True, ssl_trusted_ca: Optional[str] = None,
                ssl_client_certificate: Optional[str] = None,
-               ssl_private_key: Optional[str] = None) -> "LanguageContainerDeployer":
+               ssl_private_key: Optional[str] = None,
+               extract_timeout: timedelta = timedelta(seconds=10),
```
Comment (tkilias) [inline exasol/python_extension_common/deployment/language_container_deployer.py:314]: A proper value for a timeout is beyond 10 minutes

File: exasol/python_extension_common/deployment/language_container_deployer_cli.py [hunk 2]
```diff
@@ -152,6 +153,9 @@ def secret_callback(ctx: click.Context, param: click.Option, value: Any):
 @click.option('--alter-system/--no-alter-system', type=bool, default=True)
 @click.option('--allow-override/--disallow-override', type=bool, default=False)
 @click.option('--wait_for_completion/--no-wait_for_completion', type=bool, default=True)
+@click.option('--extract-timeout-minutes', type=int, default=5)
```
Comment (tkilias) [inline exasol/python_extension_common/deployment/language_container_deployer_cli.py:156]: I would set the default at least to 10 minutes

File: exasol/python_extension_common/deployment/language_container_deployer.py [hunk 3]
```diff
@@ -87,6 +87,10 @@ def get_udf_path(bucket_base_path: bfs.path.PathLike, bucket_file: str) -> PureP
     return PurePosixPath(file_path.as_udf_path())
 
 
+def display_extract_progress(n: int, pending: List[int]):
```
Comment (tkilias) [inline exasol/python_extension_common/deployment/language_container_deployer.py:90]: How often would this be called?

File: exasol/python_extension_common/deployment/language_container_deployer.py (hunk 1 above)
Comment (ckunki) [reply exasol/python_extension_common/deployment/language_container_deployer.py:314]: Which value do you propose?

File: exasol/python_extension_common/deployment/language_container_deployer_cli.py (hunk 2 above)
Comment (ckunki) [reply exasol/python_extension_common/deployment/language_container_deployer_cli.py:156]: See next push.

File: exasol/python_extension_common/deployment/language_container_deployer.py (hunk 1 above)
Comment (ckunki) [reply exasol/python_extension_common/deployment/language_container_deployer.py:314]: I will set it to 10 minutes, see next push.

File: exasol/python_extension_common/deployment/language_container_deployer.py (hunk 3 above)
Comment (ckunki) [reply exasol/python_extension_common/deployment/language_container_deployer.py:90]: It depends on the settings. 

Default is extract-timeout-minutes (currently 10 minutes, but under discussion) 
and the interval (currently, always set to 10 secs, not configurable via CLI). 

So with these setting, by default the callback will be called 10*60/10 = 60 times at max.

Comment (tkilias) [reply exasol/python_extension_common/deployment/language_container_deployer.py:90]: I probably, would increase the interval to 30 seconds

**Comment by ahsimb** [comment]
I am not sure if the term `extract` would make sense to the user. This is a kind of knowledge bias on our side. Could `deploy-timeout-minutes` be a better name?

**Comment by tkilias** [comment]
> I am not sure if the term `extract` would make sense to the user. This is a kind of knowledge bias on our side. Could `deploy-timeout-minutes` be a better name?

Yes, I think that is a good idea

File: exasol/python_extension_common/deployment/language_container_deployer.py (hunk 3 above)
Comment (ckunki) [reply exasol/python_extension_common/deployment/language_container_deployer.py:91]: Thanks - changes in latest push.

**Review by tkilias — DISMISSED** [review]

**Review by ahsimb — APPROVED** [review]

**Pull request merged**